### ISSUE TITLE
New OWL imports cause reasoners to fail building the class hierarchy

### ISSUE THREAD

**Issue by flekschas** [obophenotype/cell-ontology#429]
**The issue:** I am working with OWLAPI `v3.5.4` and Hermit (`v1.3.8.4`) as well as Protege (`v5-beta17`). Up until a few weeks ago I was able to build the class hierarchy of CL without any issues. Now (without having changed my code at all) building the class hierarchy leads to an "out of memory". I am able to reproduce this with my custom cli application as well as with Protege (using Hermit and Fact++).

**The cause:** I figured out that the cause for this issue are the new OWL import ontologies introduced in dd3f78b394f1e833e9fc13145750aa7a0fcc56c2.

The issue is not present in Protege `v5-beta23` and Hermit `v1.3.8.413` but building the class hierarchy is _significantly_ slower than before.

Any ideas what could cause this issue?

tl/dr:
It would also be great if the OWL imports are versioned as well.
